x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -240,6 +239,17 @@ public void afterBulk(long executionId, BulkRequest request, Throwable failure)
             .build();
     }
 
+    private record FieldValueHolder(AbstractRollupFieldProducer fieldProducer, LeafFieldData leafFieldData) {
+
+        public AbstractRollupFieldProducer getRollupFieldProducer() {
+            return fieldProducer;
+        }
+
+        public LeafFieldData getLeafFieldData() {
+            return leafFieldData;
+        }
```

Comment:
Those are probably not needed. 

Records expose their own getters `fieldProducer()` and `leafFieldData()`